modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientExperimentalCodegen.java

```diff
@@ -83,16 +82,23 @@ public void processOpts() {
                 }
 
                 for (CodegenProperty param : model.vars) {
-                    if (!param.isNullable) {
+                    if (!param.isNullable || param.isMapContainer || param.isListContainer) {
                         continue;
                     }
-
-                    param.dataType = "Nullable" + Character.toUpperCase(param.dataType.charAt(0))
+                    if (param.isDateTime) {
+                        param.dataType = "NullableTime";
```

Comment:
Agreed with using typeMapping to directly map the type instead of using the code above.

Reply:
So interestingly, adding typeMapping.put("DateTime", "NullableTime") does not produce the same output for the samples. I wonder if that was done on purpose. When the dataType is set in postProcessModels(), it does not have the same effect as when typeMapping is invoked in processOpts(). Is that done on purpose?

Reply:
We only want to represent the DateTime object as NullableTime if it's marked as nullable in the spec. I don't think we should use it in all cases.

Reply:
For example, at https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go-experimental/api.mustache#L26

```
type api{{operationId}}Request struct {
	ctx _context.Context
	apiService *{{classname}}Service{{#allParams}}
	{{paramName}} {{^isPathParam}}*{{/isPathParam}}{{{dataType}}}{{/allParams}}
}
```

With "typeMapping", this will produce  NullableTime.
With the dataType changed in processOpts(), this will produce *time.Time.

Reply:
Yeah, I agree this PR is ok as-is, let's keep more changes for followup PRs.

Reply:
> We only want to represent the DateTime object as NullableTime if it's marked as nullable in the spec. I don't think we should use it in all cases.

Thanks for the explanation. I have added code comments to explain.

Reply:
> Yeah, I agree this PR is ok as-is, let's keep more changes for followup PRs.

Thanks. Can you approve the PR then? Approving in the text does not help much. BTW, for some reason I am unable to add you as a reviewer.

Reply:
Approved. Sorry, I keep forgetting doing the official review.